packages/desktop-client/src/components/reports/Reports.js

```diff
@@ -0,0 +1,454 @@
+          <strong>How is cash flow calculated?</strong>
+        </P>
+      );
+    }
+
+    if (isIE) {
+      return (
+        <P>
+          <strong>How are income and expenses calculated?</strong>
+        </P>
+      );
+    }
+  })();
+
+  const spreadSheet = (() => {
+    return isCashFlow
+      ? 'cash-flow'
+      : isNetWorth
+      ? 'net-worth'
+      : isIE && 'ie-sheet';
+  })();
+
+  const params = useMemo(
+    () =>
+      masterDataSpreadsheet(
+        start,
+        end,
+        endDay,
+        isTotals,
+        isConcise,
+        selectList,
+        isCashFlow,
+        isNetWorth,
+        isIE,
+        filt,
+        categories,
+      ),
+    [
+      start,
+      end,
+      endDay,
+      isTotals,
+      isConcise,
+      selectList,
+      isCashFlow,
+      isNetWorth,
+      isIE,
+      filt,
+      categories,
+    ],
+  );
+  const data = useReport(spreadSheet, params);
+
+  useEffect(() => {
+    async function run() {
+      const trans = await send('get-earliest-transaction');
+      const earliestMonth = trans
+        ? monthUtils.monthFromDate(d.parseISO(trans.date))
+        : monthUtils.currentMonth();
+
+      const allMonths = monthUtils
+        .rangeInclusive(earliestMonth, monthUtils.currentMonth())
+        .map(month => ({
+          name: month,
+          pretty: monthUtils.format(month, 'MMMM, yyyy'),
+        }))
+        .reverse();
+
+      setAllMonths(allMonths);
+    }
+    run();
+  }, []);
+
+  function onSelection(onSelection) {
+    setIsCashFlow(false);
+    setIsNetWorth(false);
+    setIsIE(false);
+    onSelection(true);
+  }
+
+  function onDeleteFilter(filter) {
+    applyFilters(filterz.filter(f => f !== filter));
+  }
+
+  function deleteAllFilters() {
+    applyFilters([]);
+  }
+
+  async function onApplyFilter(cond) {
+    let filters = filterz;
+    applyFilters([...filters, cond]);
+  }
+
+  async function applyFilters(conditions) {
+    if (conditions.length > 0) {
+      let { filters } = await send('make-filters-from-conditions', {
+        conditions,
+      });
+
+      let filte = [...filters];
+      setFilt(filte);
+      setFilterz(conditions);
+    } else {
+      setFilt([]);
+      setFilterz(conditions);
+    }
+  }
+
+  function onChangeDates(start, end) {
+    const numDays = d.differenceInCalendarDays(
+      d.parseISO(end),
+      d.parseISO(start),
+    );
+    const isConcise = numDays > 31 * 3;
+
+    let endDay = end + '-31';
+    if (endDay > monthUtils.currentDay()) {
+      endDay = monthUtils.currentDay();
+    }
+
+    setStart(start + '-01');
+    setEndDay(endDay);
+    setIsConcise(isNetWorth || isIE ? true : isConcise);
+  }
+
+  if (!allMonths || !data) {
+    return null;
+  }
+
+  const handleChange = e => {
+    setSelectList(e.target.value);
+  };
+
+  const handleClick = e => {
+    if (
+      e.target.textContent === 'Totals' ||
+      e.target.id === 'TotalsChoice' ||
+      e.target.parentElement.id === 'TotalsChoice'
```

Comment:
Would it be possible to remove direct DOM access here in favor of passing the appropriate information into the event handler directly?

Reply:
Done.